### ISSUE TITLE
Upgrade to pest 2.0+

### ISSUE THREAD

**Issue by cameel** [leonardoalt/yultsur#33]
Pest 2 has some nice improvements over version 1. For example the [`as_span()` method I could have used in my PR](https://github.com/leonardoalt/yultsur/pull/32#discussion_r979317642).

**Comment by leonardoalt** [comment]
@cameel do you wanna upgrade then and change that PR to use the new stuff from pest 2?

**Comment by cameel** [comment]
I can try but later - I needed to do some stuff around yul-lsp first so I only created an issue for now.

**Comment by leonardoalt** [comment]
Should i merge the other PR now then? And then do the pest2 thing later

**Comment by cameel** [comment]
Yes. I mean, as long as you're fine with tests there - the source locations are not well covered.
